todo_app/test_app.py

```diff
@@ -11,43 +13,29 @@ def client():
     file_path = find_dotenv('.env.test')
     load_dotenv(file_path, override=True)
 
-    # Create the new app.
-    test_app = app.create_app()
+    with mongomock.patch(servers=(('fakemongo.com', 27017),)):
+        # Create the new app.
+        test_app = app.create_app()
 
-    # Use the app to create a test_client that can be used in our tests.
-    with test_app.test_client() as client:
-        yield client
+        # Use the app to create a test_client that can be used in our tests.
+        with test_app.test_client() as client:
+            yield client
 
-
-class StubResponse():
-    def __init__(self, fake_response_data):
-        self.fake_response_data = fake_response_data
-
-    def json(self):
-        return self.fake_response_data
-
-
-def stub(action, url, headers={},  params={}):
-    test_board_id = os.environ.get('BOARD_ID')
-
-    if url == f'https://api.trello.com/1/boards/{test_board_id}/lists':
-        fake_response_data = [{
-            'id': '123abc',
-            'name': 'To Do',
-            'cards': [{'id': '456', 'name': 'Test card', 'desc': 'Test desc'}]
-        }]
-        return StubResponse(fake_response_data)
-
-    raise Exception(f'Integration test did not expect URL "{url}"')
-
-
-def test_index_page(monkeypatch, client):
+def test_index_page(client):
     # Arrange
-    monkeypatch.setattr(requests, 'request', stub)
+    mongo_client = pymongo.MongoClient(os.environ.get('MONGOBD_PRIMARY_CONNECTION_STRING'))
+    db = mongo_client[os.environ.get('MONGO_DATABASE_NAME')]
+    collection = db[os.environ.get('MONGO_COLLECTION_NAME')]
+
+    collection.insert_one({
```

Comment:
Not a suggestion but a reflection - would this be easier/better if we leaned on the existing `add_item` method in your MongoService?

Pros:
* Probably easier / less code here
* Will more easily be caught/updated if we change our data persistence methods
* Updates automatically if we change the data structure of an item

Cons:
* The test is less well targeted, there are *two* good reasons it might fail
* We have less control over the specific form of the data in the DB

I don't lean particularly either way, but I can easily imagine teams having strong opinions!